### ISSUE TITLE
Matrix language tags are just working for rows

### ISSUE THREAD

**Issue by MarkusFrBoden** [surveyjs/survey-library#7654]
I have created the following matrix in my JSON. When I change the language, the row labels are automatically changed. However, the columns do not switch between English and German.

{
        "type": "matrix",
        "name": "question16",
        "title": {
         "de": "In welchen der folgenden operativen Bereiche hat Ihre Einrichtung bereits in die Digitalisierung investiert, und in welchen Bereichen plant sie, in Zukunft zu investieren? Bitte wählen Sie alle zutreffenden Optionen aus.",
         "en":"In which of the following operational areas has your organisation already invested in digitalisation and in which ones does it plan to invest in the future? Please select all options that apply.",
        },
        "columns": [
         {
          "value": "Column 1",
          "text": {
           "de": "Investition getätigt",
           "en":"Already invested",
          }
         },
         {
          "value": "Column 2",
          "text": {
           "de": "Investition geplant",
           "en":"Plan to invest",
          }
         },
         {
          "value": "Column 3",
          "text": {
           "de": "nichts derbeiden",
           "en":"nothing",
          }
         }
        ],
        "rows": [
         {
          "value": "Row 1",
          "text": {
           "de": "Interne Abläufe (administrative, strukturelle, organisatorische Prozesse)",
           "en":"Internal operations (administrative, structural, organisational processes)",
          }
         },
         {
          "value": "Row 2",
          "text": {
           "de": "Externe Maßnahmen (Governance-Prozesse einschließlich neuer Formen der Beteiligung von Bürgerinnen und Bürgern)",
           "en":"External operations (governance processes including new ways of citizen participation)",
          }
         }
        ]
       }

**Comment by andrewtelnov** [comment]
@MarkusFrBoden I have just tried your JSON on our website and it works as expected:
```js
{
 "pages": [
  {
   "name": "page1",
   "elements": [
    {
     "type": "matrix",
     "name": "question16",
     "title": {
      "de": "In welchen der folgenden operativen Bereiche hat Ihre Einrichtung bereits in die Digitalisierung investiert, und in welchen Bereichen plant sie, in Zukunft zu investieren? Bitte wählen Sie alle zutreffenden Optionen aus.",
      "en": "In which of the following operational areas has your organisation already invested in digitalisation and in which ones does it plan to invest in the future? Please select all options that apply."
     },
     "columns": [
      {
       "value": "Column 1",
       "text": {
        "de": "Investition getätigt",
        "en": "Already invested"
       }
      },
      {
       "value": "Column 2",
       "text": {
        "de": "Investition geplant",
        "en": "Plan to invest"
       }
      },
      {
       "value": "Column 3",
       "text": {
        "de": "nichts derbeiden",
        "en": "nothing"
       }
      }
     ],
     "rows": [
      {
       "value": "Row 1",
       "text": {
        "de": "Interne Abläufe (administrative, strukturelle, organisatorische Prozesse)",
        "en": "Internal operations (administrative, structural, organisational processes)"
       }
      },
      {
       "value": "Row 2",
       "text": {
        "de": "Externe Maßnahmen (Governance-Prozesse einschließlich neuer Formen der Beteiligung von Bürgerinnen und Bürgern)",
        "en": "External operations (governance processes including new ways of citizen participation)"
       }
      }
     ]
    }
   ]
  }
 ]
}
```
Could you provide an example to reproduce the issue?

Thank you,
Andrew

**Comment by MarkusFrBoden** [comment]
Hallo Andrew,

I was already thinking that I was running into some kind of special 
problem. I'm also just starting with vue frontend development so please 
don't be surprised about duplicate functions.


I use SurveyJS in my Vue component EUSurvey.vue:
------------------------------------
<template>
   <div>
     <SurveyComponent :model="survey" :darkmode="darkmode" 
:language="language"/>
   </div>
</template>

<script>
import 'survey-core/defaultV2.min.css';
import "survey-core/i18n/german";
import { Model } from 'survey-core';
import { DefaultDark } from '../styles/default-dark';
import { DefaultLight } from '../styles/default-light';
import { onMounted } from 'vue';
import { EUSurveyJSON } from "./EUSurvey_json";

export default {
   props: ['darkmode', 'language'],
   emits: [],
   data() {
     return {
       survey: null,
     };
   },
   methods: {
     handleDarkmodeChange(newDarkmode) {
       if (this.survey) {
         if (newDarkmode) {
           this.survey.applyTheme(DefaultLight);
         } else {
           this.survey.applyTheme(DefaultDark);
         }
       }
     },
     handleLanguageChange() {
         if (this.language === "en") {
           this.survey.locale = "en"
         } else {
           this.survey.locale = "de"
         }
       }
   },
   setup(props) {
     const surveyJson = EUSurveyJSON;

     const survey = new Model(surveyJson);

     onMounted(() => {
       if (props.darkmode) {
         survey.applyTheme(DefaultLight);
       } else {
         survey.applyTheme(DefaultDark);
       }

       if (props.language === "en") {
         survey.locale = "en"
       } else {
         survey.locale = "de"
       }

       survey.onComplete.add(alertResults);
     });

     const alertResults = (sender) => {
       const results = JSON.stringify(sender.data);
       alert(results);
     };

     return { survey };
   },
   watch: {
     darkmode: 'handleDarkmodeChange',
     language: 'handleLanguageChange',
   },
};
</script>

<style>
  .sd-question__header {
       text-align: left !important;
     }
   .sd-page__description {
       text-align: left !important;
     }
   .sd-panel__description {
       text-align: left !important;
     }
   .sd-panel__title {
       text-align: left !important;
     }
   .sd-item {
     text-align: left;
   }
</style>
-------------------------------------------------

The EUSurveyJSON is still the JSON that I sent:

-------------------------------------------------
export const EUSurveyJSON = {
     "title": {
         "de": "Fragebogen zur Bewertung der digitalen Reife für 
EDIH-Kunden",
         "en": "Digital Maturity Assessment Questionnaire for EDIH 
Customers"
     },
     "description": {
         "de": "Zielgruppe: öffentliche Einrichtungen Phase: T0 (vor dem 
Start der EDIH-Unterstützung)",
         "en": "Target group: Public Sector Organisations Stage: T0 
(prior to EDIH support start)"
     },
     "completedHtml": {
         "de": "<h3>Danke für das ausfüllen</h3>",
         "en": "<h3>Thank you for completing the survey</h3>"
     },
     "showQuestionNumbers": "off",
     "showProgressBar": "top",
     "showPreviewBeforeComplete": "showAllQuestions",
     "widthMode": "static",
     "width": "900",
     "fitToContainer": true,
     "pages": [
         [...]
         {
             "name": "page3",
             "elements": [
                 {
                     "type": "checkbox",
                     "name": "question17",
                     "title": {
                         "de": "Welche Vorkehrungen wurden getroffen, um 
Ihre Einrichtung auf die (weitere) Digitalisierung vorzubereiten? Bitte 
wählen Sie alle zutreffenden Optionen aus",
                         "en": "In which of the following ways is your 
organisation prepared for (more) digitalisation? Please select all 
options that apply"
                     },
                     "choices": [
                         {
                             "value": "Item 1",
                             "text": {
                                 "de": "Der Digitalisierungsbedarf wurde 
ermittelt und ist auf die politischen/administrativen Zielen 
abgestimmt",
                                 "en": "Digitalisation needs are 
identified and are aligned with policy making/administrative objectives"
                             }
                         },
                         {
                             "value": "Item 2",
                             "text": {
                                 "de": "Die finanziellen Ressourcen 
(Eigenmittel, Zuschüsse, Darlehen), mit denen die 
Digitalisierungsprojekte mindestens in den folgenden zwei Jahren 
abgesichert sind, wurden ermittelt",
                                 "en": "Financial resources (own funds, 
grants, loans) are identified to secure digitalisation projects during 
at least the following two years"
                             }
                         },
                         {
                             "value": "Item 3",
                             "text": {
                                 "de": "Die IKT-Infrastrukturen sind für 
eine Unterstützung der Digitalisierungspläne bereit",
                                 "en": "ICT infrastructures are ready to 
support digitalisation plans"
                             }
                         }
                     ]
                 },
                 {
                     "type": "matrix",
                     "name": "question16",
                     "title": {
                         "de": "In welchen der folgenden operativen 
Bereiche hat Ihre Einrichtung bereits in die Digitalisierung investiert, 
und in welchen Bereichen plant sie, in Zukunft zu investieren? Bitte 
wählen Sie alle zutreffenden Optionen aus.",
                         "en": "In which of the following operational 
areas has your organisation already invested in digitalisation and in 
which ones does it plan to invest in the future? Please select all 
options that apply."
                     },
                     "columns": [
                         {
                             "value": "Column 1",
                             "text": {
                                 "de": "Investition getätigt",
                                 "en": "Already invested"
                             }
                         },
                         {
                             "value": "Column 2",
                             "text": {
                                 "de": "Investition geplant",
                                 "en": "Plan to invest"
                             }
                         },
                         {
                             "value": "Column 3",
                             "text": {
                                 "de": "nichts derbeiden",
                                 "en": "nothing"
                             }
                         }
                     ],
                     "rows": [
                         {
                             "value": "Row 1",
                             "text": {
                                 "de": "Interne Abläufe (administrative, 
strukturelle, organisatorische Prozesse)",
                                 "en": "Internal operations 
(administrative, structural, organisational processes)"
                             }
                         },
                         {
                             "value": "Row 2",
                             "text": {
                                 "de": "Externe Maßnahmen 
(Governance-Prozesse einschließlich neuer Formen der Beteiligung von 
Bürgerinnen und Bürgern)",
                                 "en": "External operations (governance 
processes including new ways of citizen participation)"
                             }
                         }
                     ]
                 }
             ]

         }
     ]
}
------------------------------------------

Interestingly, the language change works both when loading the page and 
when changing the language without reloading all other elements of the 
survey. For the columns, however, only the function in the lifecycle 
hook onMounted works, so when I set the language and the page is 
reloaded, but not when I change the language on an already loaded page.

What doesn't change either is the language of the UI, for example of the 
date input fields, but that's probably a different topic.

Thanks for the support, we are strongly considering buying SurveyJ's 
licenses for the university.

Kind regards
Markus Boden


Am 2024-01-11 21:34, schrieb Andrew:

**Comment by andrewtelnov** [comment]
@MarkusFrBoden What version do you use?
[Here is the example ](https://codesandbox.io/p/sandbox/surveyjs-library-vue3-forked-3rd4s4?file=%2Fsrc%2FApp.vue%3A12%2C4) with the latest version for Vue3
[Here the example](https://plnkr.co/edit/DUMQJchvcBRevcZr) with the latest version for Vue2.
Feel free to modify these examples to let us reproduce the issue. They use your JSON.

Thank you,
Andrew

**Comment by MarkusFrBoden** [comment]
Many thanks for the links. This enabled me to identify the problems. The 
Vue3 example still didn't worked for me.
I then opened it in a different browser. It worked in Chrome, but not in 
Edge. I checked the Edge extensions and found out that the Citavi 
citation software plugin was causing the problems. If I switch this off 
and now it also works here.

Thanks a lot for the help, it's really fun to work with SurveyJS!

Am 2024-01-14 10:39, schrieb Andrew:
